cookbooks/aws-parallelcluster-slurm/recipes/config/login_nodes_daemon_service.rb

```diff
@@ -0,0 +1,54 @@
+require 'yaml'
+config = YAML.safe_load(File.read(node['cluster']['cluster_config_path']))
+GRACETIME_PERIOD = config["LoginNodes"]["Pools"][0]["GRACETIME_PERIOD"]
+
+# Create configuration file
+template "#{node['cluster']['shared_dir']}/login_nodes_daemon_config.json" do
+  source 'login_nodes_daemon_service/login_nodes_daemon_config.json.erb'
+  owner 'root'
+  group 'root'
+  mode '0644'
+  variables(
+    {
+      GRACETIME_PERIOD: GRACETIME_PERIOD,
+    }
+  )
+end
+
+# Create on_termination_script.sh
+template "#{node['cluster']['shared_dir']}/login_nodes_on_termination_script.sh" do
+  source 'login_nodes_daemon_service/login_nodes_on_termination_script.sh.erb'
+  owner 'root'
+  group 'root'
+  mode '0755'
+end
+
+# Create daemon_script.sh
+template "#{node['cluster']['shared_dir']}/login_nodes_daemon_script.sh" do
+  source 'login_nodes_daemon_service/login_nodes_daemon_script.sh.erb'
+  owner 'root'
+  group 'root'
+  mode '0755'
+end
+
+# Create supervisord configuration for the daemon
+file "/etc/supervisor/conf.d/login_nodes_daemon_service.conf" do
+  content <<-EOF
+[program:login_nodes_daemon_service]
+command=bash #{node['cluster']['shared_dir']}/login_nodes_daemon_script.sh #{GRACETIME_PERIOD}
+autostart=true
```

Comment:
The autorestart policy strictly depends on the way the program has been implemented. 
In this specific case, the program should not be auto-restarted if it exits with exit code 0, otherwise it would send multiple notifications.

So we have two options here:
1. leave the daemon logic as it is, but make supervisord restart the process only if it exits with exti code != 0
2. leave supervisord always restarting the process as it currently does, but change the daemon logic in a way that it avoids sending multiple notifications

Reply:
I use the first option to do the change:
leave the daemon logic as it is, but make supervisord restart the process only if it exits with exti code != 0

Reply:
We need the following configurations:

```
exitcodes = 0
autostart = true
autorestart = unexpected
```

With autorestart = unexpected, supervisord will restart the daemon only if it exits with unexpected exit codes, i.e. in this case code != 0.